doc/development/guide/documentation.rst

```diff
@@ -1,39 +1,449 @@
+  information, however will be displayed in full to users in the terminal/Jupyter notebooks, so
+  do not let this section become too long.
+
+
+Classes
+~~~~~~~
+
+The class docstring is placed directly below the class definition:
+
+.. code-block:: python
+
+    class MyClass:
+        """Single sentence that summarizes the class.
+
+        Multi-line description of the class (optional, if required).
+
+        Args:
+            arg1 (type): Description.
+                Continuation line is indented if needed.
+            arg2 (type): description
+
+        Keyword Args:
+            kwarg1 (type): description
+
+        Attributes:
+            attr1 (type): description
+
+        Raises:
+            ExceptionType: description
+
+        .. seealso:: :func:`~.relevant_func`, :class:`~.RelevantClass` (optional)
+
+        **Example**
+
+        Minimal example with 1 or 2 code blocks (required).
+
+        .. UsageDetails::
+
+            More complicated use cases, options, and larger code blocks (optional).
+
+        **Related tutorials**
+
+        Links to any relevant PennyLane demos (optional).
+        """
+
+Docstrings for classes follow a similar structure as functions, with a few differences:
+
+* **Include initialization arguments.** The constructor ``__init__()`` is documented
+  here. Note that as the constructor simply returns an instance of the class, no ``Returns:``
+  section is included.
+
+* **Do not list methods and properties.** Class properties and methods will be automatically
+  listed alongside the class in the generated documentation.
+
+* **List attributes if relevant.**
+
+Document all methods and properties as you would for functions above. However, there are two
+exemptions, where docstrings should *not* be provided:
+
+* **Magic or special methods.** This includes methods such as ``__init__``, ``__call__``,
+  ``__len__``, ``__getitem__``, etc. Only provide docstrings for special methods if
+  their behaviour is modified in a non-standard way (e.g., if negative indices are allowed
+  as arguments for ``__getitem__``).
+
+* **Overwritten methods.** Overwritten inherited methods automatically inherit the parent methods
+  docstring, even when overwritten. This is particularly useful when creating an instance
+  of an abstract base class; the abstract method docstring is defined in the parent, and automatically
+  inherited by the child class, even when overwritten.
+
+  As with special methods, only provide docstrings for overwritten methods if
+  their behaviour is modified in a non-standard way (as defined in the parent class).
+
+
+Variables
+~~~~~~~~~
+
+Module level variables may also be optionally documented, by providing a triple-quote docstring
+**below** the variable definition. For example,
+
+.. code-block:: python
+
+    x = {"John": 23, "James": 54}
+    """dict[str, int]: stores the ages of known users"""
```

Comment:
I've never seen this done in PL codebase for module-level variables, but I've seen it done for class attributes

Reply:
There's a weird quirk in sphinx where these are automatically extracted for module level variables, but not class attributes :( so the ones we have for class variables are not appearing in the docs if the `Attributes` section is not provided.